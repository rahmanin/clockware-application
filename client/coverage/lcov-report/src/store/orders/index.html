
<!doctype html>
<html lang="en">

<head>
    <title>Code coverage report for src/store/orders</title>
    <meta charset="utf-8" />
    <link rel="stylesheet" href="../../../prettify.css" />
    <link rel="stylesheet" href="../../../base.css" />
    <meta name="viewport" content="width=device-width, initial-scale=1">
    <style type='text/css'>
        .coverage-summary .sorter {
            background-image: url(../../../sort-arrow-sprite.png);
        }
    </style>
</head>
    
<body>
<div class='wrapper'>
    <div class='pad1'>
        <h1><a href="../../../index.html">All files</a> src/store/orders</h1>
        <div class='clearfix'>
            
            <div class='fl pad1y space-right2'>
                <span class="strong">88.64% </span>
                <span class="quiet">Statements</span>
                <span class='fraction'>39/44</span>
            </div>
        
            
            <div class='fl pad1y space-right2'>
                <span class="strong">76.92% </span>
                <span class="quiet">Branches</span>
                <span class='fraction'>10/13</span>
            </div>
        
            
            <div class='fl pad1y space-right2'>
                <span class="strong">76.92% </span>
                <span class="quiet">Functions</span>
                <span class='fraction'>10/13</span>
            </div>
        
            
            <div class='fl pad1y space-right2'>
                <span class="strong">90.24% </span>
                <span class="quiet">Lines</span>
                <span class='fraction'>37/41</span>
            </div>
        
            
        </div>
        <p class="quiet">
            Press <em>n</em> or <em>j</em> to go to the next uncovered block, <em>b</em>, <em>p</em> or <em>k</em> for the previous block.
        </p>
    </div>
    <div class='status-line high'></div>
    <div class="pad1">
<table class="coverage-summary">
<thead>
<tr>
   <th data-col="file" data-fmt="html" data-html="true" class="file">File</th>
   <th data-col="pic" data-type="number" data-fmt="html" data-html="true" class="pic"></th>
   <th data-col="statements" data-type="number" data-fmt="pct" class="pct">Statements</th>
   <th data-col="statements_raw" data-type="number" data-fmt="html" class="abs"></th>
   <th data-col="branches" data-type="number" data-fmt="pct" class="pct">Branches</th>
   <th data-col="branches_raw" data-type="number" data-fmt="html" class="abs"></th>
   <th data-col="functions" data-type="number" data-fmt="pct" class="pct">Functions</th>
   <th data-col="functions_raw" data-type="number" data-fmt="html" class="abs"></th>
   <th data-col="lines" data-type="number" data-fmt="pct" class="pct">Lines</th>
   <th data-col="lines_raw" data-type="number" data-fmt="html" class="abs"></th>
</tr>
</thead>
<tbody><tr>
	<td class="file medium" data-value="actions.ts"><a href="actions.ts.html">actions.ts</a></td>
	<td data-value="73.68" class="pic medium">
	<div class="chart"><div class="cover-fill" style="width: 73%"></div><div class="cover-empty" style="width: 27%"></div></div>
	</td>
	<td data-value="73.68" class="pct medium">73.68%</td>
	<td data-value="19" class="abs medium">14/19</td>
	<td data-value="50" class="pct medium">50%</td>
	<td data-value="2" class="abs medium">1/2</td>
	<td data-value="70" class="pct medium">70%</td>
	<td data-value="10" class="abs medium">7/10</td>
	<td data-value="77.78" class="pct medium">77.78%</td>
	<td data-value="18" class="abs medium">14/18</td>
	</tr>

<tr>
	<td class="file high" data-value="reducer.ts"><a href="reducer.ts.html">reducer.ts</a></td>
	<td data-value="100" class="pic high">
	<div class="chart"><div class="cover-fill cover-full" style="width: 100%"></div><div class="cover-empty" style="width: 0%"></div></div>
	</td>
	<td data-value="100" class="pct high">100%</td>
	<td data-value="25" class="abs high">25/25</td>
	<td data-value="81.82" class="pct high">81.82%</td>
	<td data-value="11" class="abs high">9/11</td>
	<td data-value="100" class="pct high">100%</td>
	<td data-value="3" class="abs high">3/3</td>
	<td data-value="100" class="pct high">100%</td>
	<td data-value="23" class="abs high">23/23</td>
	</tr>

</tbody>
</table>
</div>
                <div class='push'></div><!-- for sticky footer -->
            </div><!-- /wrapper -->
            <div class='footer quiet pad2 space-top1 center small'>
                Code coverage generated by
                <a href="https://istanbul.js.org/" target="_blank">istanbul</a>
                at Tue Dec 01 2020 12:22:24 GMT+0200 (Eastern European Standard Time)
            </div>
        </div>
        <script src="../../../prettify.js"></script>
        <script>
            window.onload = function () {
                prettyPrint();
            };
        </script>
        <script src="../../../sorter.js"></script>
        <script src="../../../block-navigation.js"></script>
    </body>
</html>
    