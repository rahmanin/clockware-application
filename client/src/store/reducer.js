import { combineReducers } from "redux";
import pricesReducer from "./prices/reducer";

const rootReducer = combineReducers({
  prices: pricesReducer,
});

export default rootReducer;