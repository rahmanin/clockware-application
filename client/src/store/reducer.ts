import { combineReducers } from "redux";
import pricesReducer from "./prices/reducer";
import usersReducer from "./users/reducer";
import citiesReducer from "./cities/reducer";
import mastersReducer from "./masters/reducer";
import ordersReducer from "./orders/reducer";
import ordersClientReducer from "./ordersClient/reducer";
import languageReducer from "./language/reducer";
import { createStore, applyMiddleware } from "redux";
import thunk from 'redux-thunk';
import { composeWithDevTools } from 'redux-devtools-extension'

const rootReducer = combineReducers({
  prices: pricesReducer,
  user: usersReducer,
  cities: citiesReducer,
  masters: mastersReducer,
  orders: ordersReducer,
  ordersClient: ordersClientReducer,
  language: languageReducer
});

const composedEnhancer = composeWithDevTools(
  applyMiddleware(thunk)
)

const store = createStore(
  rootReducer,
  composedEnhancer
);

export default store
