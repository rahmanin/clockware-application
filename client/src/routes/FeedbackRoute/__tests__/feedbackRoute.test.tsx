import React from 'react';
import {cleanup} from '@testing-library/react';
import { Provider } from "react-redux";
import store from "../../../store/reducer";
import { act } from "react-dom/test-utils";
import {mount} from "enzyme"
import Adapter from 'enzyme-adapter-react-16';
import Enzyme from 'enzyme';
import Feedback from '../index';
import { MemoryRouter } from 'react-router';

Enzyme.configure({ adapter: new Adapter() });
afterEach(cleanup);

Object.defineProperty(window, 'matchMedia', {
  writable: true,
  value: jest.fn().mockImplementation(query => ({
    matches: false,
    media: query,
    onchange: null,
    addListener: jest.fn(), // deprecated
    removeListener: jest.fn(), // deprecated
    addEventListener: jest.fn(),
    removeEventListener: jest.fn(),
    dispatchEvent: jest.fn(),
  })),
});

act(() => {
  it('Render feedback form', () => {
    const component = mount(
      <Provider store={store}>
        <MemoryRouter initialEntries={[`/feedback?token=aaaaaaaaaaaaaaa&order={%22size%22:%22Small%22,%22city%22:%22Uzhhorod%22,%22order_date%22:%222020-12-01%22,%22order_time_start%22:%2213:00%22,%22order_master%22:%22Alex%22,%22feedback_master%22:%22%22,%22order_price%22:110,%22additional_price%22:0}`]}>
          <Feedback/>
        </MemoryRouter>
      </Provider>
    );
    expect(component.debug()).toMatchSnapshot();
  });
})
