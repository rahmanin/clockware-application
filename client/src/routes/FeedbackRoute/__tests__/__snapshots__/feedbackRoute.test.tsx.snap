// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Render feedback form 1`] = `
"<Provider store={{...}}>
  <MemoryRouter initialEntries={{...}}>
    <Router history={{...}}>
      <Feedback>
        <div className=\\"wrapper_feedback\\">
          <Card className=\\"order\\" title=\\"Order\\" style={{...}}>
            <div style={{...}} className=\\"ant-card ant-card-bordered order\\">
              <div className=\\"ant-card-head\\" style={{...}}>
                <div className=\\"ant-card-head-wrapper\\">
                  <div className=\\"ant-card-head-title\\">
                    Order
                  </div>
                </div>
              </div>
              <div className=\\"ant-card-body\\" style={{...}}>
                <p className=\\"order_content\\">
                  <span className=\\"order_header\\">
                    City: 
                  </span>
                  Uzhhorod
                </p>
                <p className=\\"order_content\\">
                  <span className=\\"order_header\\">
                    Size: 
                  </span>
                  Small
                </p>
                <p className=\\"order_content\\">
                  <span className=\\"order_header\\">
                    Date: 
                  </span>
                  2020-12-01
                </p>
                <p className=\\"order_content\\">
                  <span className=\\"order_header\\">
                    Time: 
                  </span>
                  13:00
                </p>
                <p className=\\"order_content\\">
                  <span className=\\"order_header\\">
                    Master: 
                  </span>
                  Alex
                </p>
                <p className=\\"order_content\\">
                  <span className=\\"order_header\\">
                    Master&#39;s feedback: 
                  </span>
                  No feedback
                </p>
                <p className=\\"order_content\\">
                  <span className=\\"order_header\\">
                    Additional price: 
                  </span>
                  0
                   hrn
                </p>
                <p className=\\"order_content\\">
                  <span className=\\"order_header\\">
                    Total price: 
                  </span>
                  110
                   hrn
                </p>
              </div>
            </div>
          </Card>
          <ForwardRef(InternalForm) className=\\"feedback_form\\" labelCol={{...}} wrapperCol={{...}} layout=\\"horizontal\\">
            <SizeContextProvider size={[undefined]}>
              <ForwardRef(Form) id={[undefined]} name={[undefined]} onFinishFailed={[Function: onInternalFinishFailed]} form={{...}} className=\\"ant-form ant-form-horizontal feedback_form\\">
                <form id={[undefined]} className=\\"ant-form ant-form-horizontal feedback_form\\" onSubmit={[Function: onSubmit]}>
                  <ForwardRef className=\\"form_item\\" name=\\"feedback_client\\" placeholder=\\"100 symbols max\\" disabled={false} rows={4} onChange={[Function (anonymous)]} value=\\"\\">
                    <ClearableLabeledInput name=\\"feedback_client\\" placeholder=\\"100 symbols max\\" disabled={false} rows={4} onChange={[Function (anonymous)]} value=\\"\\" prefixCls=\\"ant-input\\" direction={[undefined]} inputType=\\"text\\" element={{...}} handleReset={[Function: handleReset]} bordered={true}>
                      <TextArea name=\\"feedback_client\\" placeholder=\\"100 symbols max\\" disabled={false} rows={4} onChange={[Function: handleChange]} value=\\"\\" maxLength={[undefined]} className=\\"form_item\\" style={[undefined]} prefixCls=\\"ant-input\\">
                        <ResizableTextArea name=\\"feedback_client\\" placeholder=\\"100 symbols max\\" disabled={false} rows={4} onChange={[Function (anonymous)]} value=\\"\\" maxLength={[undefined]} className=\\"form_item\\" style={[undefined]} prefixCls=\\"ant-input\\" onKeyDown={[Function (anonymous)]}>
                          <ResizeObserver onResize={[Function (anonymous)]} disabled={true}>
                            <textarea name=\\"feedback_client\\" placeholder=\\"100 symbols max\\" disabled={false} rows={4} onChange={[Function (anonymous)]} value=\\"\\" maxLength={[undefined]} className=\\"ant-input form_item\\" style={{...}} onKeyDown={[Function (anonymous)]} />
                          </ResizeObserver>
                        </ResizableTextArea>
                      </TextArea>
                    </ClearableLabeledInput>
                  </ForwardRef>
                  <RatingStars defaultValue={1} className=\\"rating_stars\\" value={1} readOnly={false} precision={1} onChange={[Function (anonymous)]}>
                    <div className=\\"makeStyles-root-1\\">
                      <WithStyles(ForwardRef(Rating)) name=\\"evaluation\\" defaultValue={1} value={1} precision={1} readOnly={false} onChange={[Function (anonymous)]}>
                        <ForwardRef(Rating) classes={{...}} name=\\"evaluation\\" defaultValue={1} value={1} precision={1} readOnly={false} onChange={[Function (anonymous)]}>
                          <span onMouseMove={[Function: handleMouseMove]} onMouseLeave={[Function: handleMouseLeave]} className=\\"MuiRating-root\\" role={{...}} aria-label={{...}}>
                            <label className=\\"MuiRating-label\\" htmlFor=\\"evaluation-1\\">
                              <IconContainer value={1} className=\\"MuiRating-icon MuiRating-iconFilled\\">
                                <span className=\\"MuiRating-icon MuiRating-iconFilled\\">
                                  <ForwardRef(StarIcon) fontSize=\\"inherit\\">
                                    <WithStyles(ForwardRef(SvgIcon)) fontSize=\\"inherit\\">
                                      <ForwardRef(SvgIcon) classes={{...}} fontSize=\\"inherit\\">
                                        <svg className=\\"MuiSvgIcon-root MuiSvgIcon-fontSizeInherit\\" focusable=\\"false\\" viewBox=\\"0 0 24 24\\" color={[undefined]} aria-hidden={true} role={[undefined]}>
                                          <path d=\\"M12 17.27L18.18 21l-1.64-7.03L22 9.24l-7.19-.61L12 2 9.19 8.63 2 9.24l5.46 4.73L5.82 21z\\" />
                                        </svg>
                                      </ForwardRef(SvgIcon)>
                                    </WithStyles(ForwardRef(SvgIcon))>
                                  </ForwardRef(StarIcon)>
                                </span>
                              </IconContainer>
                              <span className=\\"MuiRating-visuallyhidden\\">
                                1 Star
                              </span>
                            </label>
                            <input onFocus={[Function: handleFocus]} onBlur={[Function: handleBlur]} onChange={[Function: handleChange]} onClick={[Function: handleClear]} disabled={false} value={1} id=\\"evaluation-1\\" type=\\"radio\\" name=\\"evaluation\\" checked={true} className=\\"MuiRating-visuallyhidden\\" />
                            <label className=\\"MuiRating-label\\" htmlFor=\\"evaluation-2\\">
                              <IconContainer value={2} className=\\"MuiRating-icon MuiRating-iconEmpty\\">
                                <span className=\\"MuiRating-icon MuiRating-iconEmpty\\">
                                  <ForwardRef(StarIcon) fontSize=\\"inherit\\">
                                    <WithStyles(ForwardRef(SvgIcon)) fontSize=\\"inherit\\">
                                      <ForwardRef(SvgIcon) classes={{...}} fontSize=\\"inherit\\">
                                        <svg className=\\"MuiSvgIcon-root MuiSvgIcon-fontSizeInherit\\" focusable=\\"false\\" viewBox=\\"0 0 24 24\\" color={[undefined]} aria-hidden={true} role={[undefined]}>
                                          <path d=\\"M12 17.27L18.18 21l-1.64-7.03L22 9.24l-7.19-.61L12 2 9.19 8.63 2 9.24l5.46 4.73L5.82 21z\\" />
                                        </svg>
                                      </ForwardRef(SvgIcon)>
                                    </WithStyles(ForwardRef(SvgIcon))>
                                  </ForwardRef(StarIcon)>
                                </span>
                              </IconContainer>
                              <span className=\\"MuiRating-visuallyhidden\\">
                                2 Stars
                              </span>
                            </label>
                            <input onFocus={[Function: handleFocus]} onBlur={[Function: handleBlur]} onChange={[Function: handleChange]} onClick={[Function: handleClear]} disabled={false} value={2} id=\\"evaluation-2\\" type=\\"radio\\" name=\\"evaluation\\" checked={false} className=\\"MuiRating-visuallyhidden\\" />
                            <label className=\\"MuiRating-label\\" htmlFor=\\"evaluation-3\\">
                              <IconContainer value={3} className=\\"MuiRating-icon MuiRating-iconEmpty\\">
                                <span className=\\"MuiRating-icon MuiRating-iconEmpty\\">
                                  <ForwardRef(StarIcon) fontSize=\\"inherit\\">
                                    <WithStyles(ForwardRef(SvgIcon)) fontSize=\\"inherit\\">
                                      <ForwardRef(SvgIcon) classes={{...}} fontSize=\\"inherit\\">
                                        <svg className=\\"MuiSvgIcon-root MuiSvgIcon-fontSizeInherit\\" focusable=\\"false\\" viewBox=\\"0 0 24 24\\" color={[undefined]} aria-hidden={true} role={[undefined]}>
                                          <path d=\\"M12 17.27L18.18 21l-1.64-7.03L22 9.24l-7.19-.61L12 2 9.19 8.63 2 9.24l5.46 4.73L5.82 21z\\" />
                                        </svg>
                                      </ForwardRef(SvgIcon)>
                                    </WithStyles(ForwardRef(SvgIcon))>
                                  </ForwardRef(StarIcon)>
                                </span>
                              </IconContainer>
                              <span className=\\"MuiRating-visuallyhidden\\">
                                3 Stars
                              </span>
                            </label>
                            <input onFocus={[Function: handleFocus]} onBlur={[Function: handleBlur]} onChange={[Function: handleChange]} onClick={[Function: handleClear]} disabled={false} value={3} id=\\"evaluation-3\\" type=\\"radio\\" name=\\"evaluation\\" checked={false} className=\\"MuiRating-visuallyhidden\\" />
                            <label className=\\"MuiRating-label\\" htmlFor=\\"evaluation-4\\">
                              <IconContainer value={4} className=\\"MuiRating-icon MuiRating-iconEmpty\\">
                                <span className=\\"MuiRating-icon MuiRating-iconEmpty\\">
                                  <ForwardRef(StarIcon) fontSize=\\"inherit\\">
                                    <WithStyles(ForwardRef(SvgIcon)) fontSize=\\"inherit\\">
                                      <ForwardRef(SvgIcon) classes={{...}} fontSize=\\"inherit\\">
                                        <svg className=\\"MuiSvgIcon-root MuiSvgIcon-fontSizeInherit\\" focusable=\\"false\\" viewBox=\\"0 0 24 24\\" color={[undefined]} aria-hidden={true} role={[undefined]}>
                                          <path d=\\"M12 17.27L18.18 21l-1.64-7.03L22 9.24l-7.19-.61L12 2 9.19 8.63 2 9.24l5.46 4.73L5.82 21z\\" />
                                        </svg>
                                      </ForwardRef(SvgIcon)>
                                    </WithStyles(ForwardRef(SvgIcon))>
                                  </ForwardRef(StarIcon)>
                                </span>
                              </IconContainer>
                              <span className=\\"MuiRating-visuallyhidden\\">
                                4 Stars
                              </span>
                            </label>
                            <input onFocus={[Function: handleFocus]} onBlur={[Function: handleBlur]} onChange={[Function: handleChange]} onClick={[Function: handleClear]} disabled={false} value={4} id=\\"evaluation-4\\" type=\\"radio\\" name=\\"evaluation\\" checked={false} className=\\"MuiRating-visuallyhidden\\" />
                            <label className=\\"MuiRating-label\\" htmlFor=\\"evaluation-5\\">
                              <IconContainer value={5} className=\\"MuiRating-icon MuiRating-iconEmpty\\">
                                <span className=\\"MuiRating-icon MuiRating-iconEmpty\\">
                                  <ForwardRef(StarIcon) fontSize=\\"inherit\\">
                                    <WithStyles(ForwardRef(SvgIcon)) fontSize=\\"inherit\\">
                                      <ForwardRef(SvgIcon) classes={{...}} fontSize=\\"inherit\\">
                                        <svg className=\\"MuiSvgIcon-root MuiSvgIcon-fontSizeInherit\\" focusable=\\"false\\" viewBox=\\"0 0 24 24\\" color={[undefined]} aria-hidden={true} role={[undefined]}>
                                          <path d=\\"M12 17.27L18.18 21l-1.64-7.03L22 9.24l-7.19-.61L12 2 9.19 8.63 2 9.24l5.46 4.73L5.82 21z\\" />
                                        </svg>
                                      </ForwardRef(SvgIcon)>
                                    </WithStyles(ForwardRef(SvgIcon))>
                                  </ForwardRef(StarIcon)>
                                </span>
                              </IconContainer>
                              <span className=\\"MuiRating-visuallyhidden\\">
                                5 Stars
                              </span>
                            </label>
                            <input onFocus={[Function: handleFocus]} onBlur={[Function: handleBlur]} onChange={[Function: handleChange]} onClick={[Function: handleClear]} disabled={false} value={5} id=\\"evaluation-5\\" type=\\"radio\\" name=\\"evaluation\\" checked={false} className=\\"MuiRating-visuallyhidden\\" />
                          </span>
                        </ForwardRef(Rating)>
                      </WithStyles(ForwardRef(Rating))>
                      (
                      1
                      )
                    </div>
                  </RatingStars>
                  <Button type=\\"primary\\" onClick={[Function: formSubmit]} disabled={false} loading={false} ghost={false} block={false} htmlType=\\"button\\">
                    <Wave>
                      <button onClick={[Function: handleClick]} disabled={false} type=\\"button\\" className=\\"ant-btn ant-btn-primary\\">
                        <LoadingIcon existIcon={false} prefixCls=\\"ant-btn\\" loading={false}>
                          <CSSMotion visible={false} motionName=\\"ant-btn-loading-icon-motion\\" removeOnLeave={true} onAppearStart={[Function: getCollapsedWidth]} onAppearActive={[Function: getRealWidth]} onEnterStart={[Function: getCollapsedWidth]} onEnterActive={[Function: getRealWidth]} onLeaveStart={[Function: getRealWidth]} onLeaveActive={[Function: getCollapsedWidth]}>
                            <DomWrapper />
                          </CSSMotion>
                        </LoadingIcon>
                        <span>
                          Ok
                        </span>
                      </button>
                    </Wave>
                  </Button>
                </form>
              </ForwardRef(Form)>
            </SizeContextProvider>
          </ForwardRef(InternalForm)>
          <ToastContainer className=\\"toast\\" position=\\"top-center\\" autoClose={false} hideProgressBar={false} newestOnTop={false} closeOnClick={true} rtl={false} pauseOnFocusLoss={true} draggable={true} pauseOnHover={true} transition={[Function: ToastTransition]} closeButton={[Function: CloseButton]} draggablePercent={80} role=\\"alert\\">
            <div className=\\"Toastify\\" id={[undefined]} />
          </ToastContainer>
        </div>
      </Feedback>
    </Router>
  </MemoryRouter>
</Provider>"
`;
